fleetctl/fleetctl.go

```diff
@@ -732,6 +803,64 @@ func lazyCreateUnits(args []string) error {
 	return nil
 }
 
+// matchUnitFiles compares two unitFiles
+// Returns true if the units match, false otherwise.
+func matchUnitFiles(a *unit.UnitFile, b *unit.UnitFile) bool {
+	if a.Hash() == b.Hash() {
+		return true
+	}
+
+	return false
+}
+
+// matchLocalFileAndUnit compares a file with a Unit
+// Returns true if the contents of the file matches the unit one, false
+// otherwise; and any error ocountered
+func matchLocalFileAndUnit(file string, su *schema.Unit) (bool, error) {
+	result := false
+	a := schema.MapSchemaUnitOptionsToUnitFile(su.Options)
+
+	_, err := os.Stat(file)
+	if err == nil {
+		b, err := getUnitFromFile(file)
+		if err == nil {
+			result = matchUnitFiles(a, b)
+		}
+	}
+
+	return result, err
+}
+
+func isLocalUnitDifferent(file string, su *schema.Unit, warnIfDifferent bool, fatal bool) (bool, error) {
```

Comment:
This might actually be the least obvious function of the bunch -- yet it has no comment at all?...

TBH, this code seems badly factored to me: between this function and it's caller, they seem to be doing closely related conditional handling, with no obvious separation of concerns. Maybe it's just me -- but I find this code very hard to follow, and nearly impossible to figure out whether the logic is actually sound...